### DISCUSSION TITLE
Compiling cmakrlist for paddle framework

### DISCUSSION THREAD

**Discussion by leeleolay** [deepmodeling/deepmd-kit#1600] [Q&A] [answered]
I am modifying the compile file <Findtensorflow.cmake>  for integrating the paddle framework, however, the cmakelist blocked my job, there is a function seems to find  the header  and Library path of tensorflow,  but to support paddle framework, could you support to add such function? I am a bit confused of this makefile and how to modify it to adapt to other framework, thanks a lot!

**Comment by PangHua** [comment]
any comment?  @njzjz

**Comment by amcadmus** [comment, marked as answer]
In your case you need to write by yourself the find paddle script. Perhaps there already exists one in the paddle community, because who want to compile with paddle with cmake would need one. In case there is not, you can find plenty of tutorials on the cmake `find_package`, some of the examples are 
https://cmake.org/cmake/help/latest/command/find_package.html
https://gitlab.kitware.com/cmake/community/-/wikis/doc/tutorials/How-To-Find-Libraries
https://github.com/rpavlik/cmake-modules/blob/main/module-docs/Example-FindMyPackage-UsingImportedTargets.cmake
https://dev.to/bruxisma/how-to-find-packages-with-cmake-the-basics-ikk
